### PR TITLE
Various little changes.

### PR DESCRIPTION
**Pull request by neuschaefer**
Some mostly independent minor fixes.

### CODE REVIEW

File: src/line.rs [hunk 1]
```diff
@@ -38,56 +38,54 @@ impl<S: BaseNum, V: Vector<S>, P: Point<S, V>>  Line<P> {
...
+        let u = cross_2 / cross_1;
 
-                return None;
-            }
+        if zero <= t && u >= zero && u <= One::one() {
+            return Some(Point2::new(p.x + t*r.x, p.y + t*r.y));
         }
+
+        return None;
     }
```
Comment (kvark) [inline src/line.rs]: Is this the final return? Should be just `None` then

**Comment by kvark** [comment]
This is super cool. Thank you @neuschaefer !
If you are going to answer/fix my small nitpick, I'll wait. Otherwise, I'd be happy to merge.

File: src/line.rs (hunk 1 above)
Comment (neuschaefer) [inline src/line.rs]: Turns out to be, yes. I'll change it in a follow-up commit.

**Comment by neuschaefer** [comment]
I'll fix it later (unless you're faster), feel free to merge!

**Comment by kvark** [comment]
Thanks a bunch!

**Pull request merged**